modules/create_timepoint/jsx/create_timepoint.js

```diff
@@ -0,0 +1,352 @@
+import React from 'react';
+import Loader from 'Loader';
+import PropTypes from 'prop-types';
+
+/**
+ * Create Timepoint.
+ *
+ * @description form for create timepoint.
+ *
+ * @author Alizée Wickenheiser
+ * @version 1.0.0
+ *
+ */
+class CreateTimepoint extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: {
+        pscid: '',
+        dccid: '',
+        visit: '',
+      },
+      form: {
+        display: {
+          subproject: false,
+          visit: false,
+          psc: false,
+        },
+        options: {
+          subproject: {},
+          visit: {},
+          psc: {},
+        },
+        value: {
+          subproject: '',
+          visit: '',
+          psc: '',
+        },
+      },
+      storage: {
+        visit: {},
+      },
+      errors: false,
+      url: {
+        params: {
+          candID: '',
+          identifier: '',
+        },
+      },
+      success: false,
+    };
+
+    // Bind component instance to custom methods
+    this.fetchInitializerData = this.fetchInitializerData.bind(this);
+    this.handleVisitLabel = this.handleVisitLabel.bind(this);
+    this.populateErrors = this.populateErrors.bind(this);
+    this.collectParams = this.collectParams.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.setForm = this.setForm.bind(this);
+  }
+  /**
+   * Executes after component mounts.
+   */
+  componentDidMount() {
+    this.collectParams();
+    this.fetchInitializerData();
+  }
+  /**
+   * Retrieve params from the browser URL and save it in state.
+   */
+  collectParams() {
+    const url = new URL(window.location.href);
+    const state = Object.assign({}, this.state);
+    state.url.params = {
+      candID: url.searchParams.get('candID'),
+      identifier: url.searchParams.get('identifier'),
+    };
+    state.data.dccid = state.url.params.candID;
+    this.setState(state);
+  }
+  /**
+   * Used with sending POST data to the server.
+   * @param {object} json - json object converted for POST.
+   * @return {string} send in POST to server.
+   */
+  urlSearchParams(json) {
+    return Object.keys(json).map((key) => {
+      return encodeURIComponent(key) + '=' + encodeURIComponent(json[key]);
+    }).join('&');
+  }
+  /**
+   * Retrieve data from the provided URL and save it in state.
+   */
+  fetchInitializerData() {
+    const state = Object.assign({}, this.state);
```

Comment:
i dont think you need to define the state const here because you do it again down below in the if statements

Reply:
I'm using the state const to access for building the json that's send to the server. I think its alight.

Reply:
oh i see where you use it just below